src/System.Management.Automation/engine/ComInterop/InteropServices/ComEventsMethod.cs

```diff
@@ -0,0 +1,277 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace System.Management.Automation.InteropServices
+{
+    /// <summary>
+    /// Part of ComEventHelpers APIs which allow binding
+    /// managed delegates to COM's connection point based events.
+    /// </summary>
+    internal class ComEventsMethod
+    {
+        /// <summary>
+        /// This delegate wrapper class handles dynamic invocation of delegates. The reason for the wrapper's
+        /// existence is that under certain circumstances we need to coerce arguments to types expected by the
+        /// delegates signature. Normally, reflection (Delegate.DynamicInvoke) handles type coercion
+        /// correctly but one known case is when the expected signature is 'ref Enum' - in this case
+        /// reflection by design does not do the coercion. Since we need to be compatible with COM interop
+        /// handling of this scenario - we are pre-processing delegate's signature by looking for 'ref enums'
+        /// and cache the types required for such coercion.
+        /// </summary>
+        public class DelegateWrapper
+        {
+            private bool _once;
+            private int _expectedParamsCount;
+            private Type?[]? _cachedTargetTypes;
```

Comment:
I have not seen this syntax before...

Reply:
This is the nullable reference type syntax. It means `the type array could be null` and `elements in the type array` could be null.